### PR TITLE
Added image reading function for runtime

### PR DESCRIPTION
**Pull request by yoya**
Added image reading function for runtime.

## Motivation and Context

In the runtime examples run.cpp,
I want to read the actual image, not the dummy random image.

## Description

Added image reading function for runtime.

```c++
  Tensor Tensor_loadImage(std::string filename);
```

We can load images in npy format even if neither link external image library.
When linking libpng, PNG can be read.
If you link opencv, you can read the format that opencv can read.

## How has this been tested?

- reference (lmnet run.py)

```shell
$ cd blueoil/output_template/python
$ python run.py -m lib_x86.so -c meta.yaml -i raw_image.png
Output: (before post process)
[[1.0594523e-04 6.2350206e-05 3.2353133e-02 3.6062470e-03 1.2402927e-02
  2.3177502e-04 9.5100403e-01 8.7061970e-05 9.8417870e-05 4.8058930e-05]]
Output:
[[1.0594523e-04 6.2350206e-05 3.2353133e-02 3.6062470e-03 1.2402927e-02
  2.3177502e-04 9.5100403e-01 8.7061970e-05 9.8417870e-05 4.8058930e-05]]
save json: output/output.json
save image: output/images/raw_image.png
```

- runtime (neither link image library)

```shell
$ cd blueoil/runtime
$ mkdir build ; cd build
$ DLK_LIB_DIR=`pwd`/../examples/dlk_lib/ cmake -DBUILD_SHARED_LIBS=ON ../
$ make install
$
$ cd ../examples
$ cp -R ../build/output/* ./
$ cmake .
$ make
$
$ ./a.out -i cat.npy -c meta.yaml
classes:
airplane
automobile
bird
cat
deer
dog
frog
horse
ship
truck
task: IMAGE.CLASSIFICATION
expected input shape:
1
32
32
3
Run
Results !
0.000105945
6.23502e-05
0.0323531
0.00360625
0.0124029
0.000231775
0.951004
8.7062e-05
9.84179e-05
4.80589e-05
```

- runtime (with libpng)

```shell
$ cd ../build/
$ rm CMakeCache.txt
$ DLK_LIB_DIR=`pwd`/../examples/dlk_lib/ cmake -DBUILD_SHARED_LIBS=ON ../ -DUSE_LIBPNG=ON
$ make install
$
$ cd ../examples
$ cp -R ../build/output/* ./
$ rm CMakeCache.txt
$ cmake . -DUSE_LIBPNG=ON
$ make
$
$ ./a.out -i cat.png -c meta.yaml
classes:
airplane
automobile
bird
cat
deer
dog
frog
horse
ship
truck
task: IMAGE.CLASSIFICATION
expected input shape:
1
32
32
3
Run
Results !
0.000105945
6.23502e-05
0.0323531
0.00360625
0.0124029
0.000231775
0.951004
8.7062e-05
9.84179e-05
4.80589e-05
```

- runtime (with opencv)

```
$ cd ../build/
$ rm CMakeCache.txt
$ DLK_LIB_DIR=`pwd`/../examples/dlk_lib/ cmake -DBUILD_SHARED_LIBS=ON ../ -DUSE_OPENCV=ON
$ make install
$
$ cd ../examples
$ cp -R ../build/output/* ./
$ rm CMakeCache.txt
$ cmake . -DUSE_OPENCV=ON
$ make
$
$ ./a.out -i cat.png -c meta.yaml
classes:
airplane
automobile
bird
cat
deer
dog
frog
horse
ship
truck
task: IMAGE.CLASSIFICATION
expected input shape:
1
32
32
3
Run
Results !
0.000105945
6.23502e-05
0.0323531
0.00360625
0.0124029
0.000231775
0.951004
8.7062e-05
9.84179e-05
4.80589e-05
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: runtime/src/blueoil_opencv.cpp [hunk 1]
```diff
@@ -20,6 +20,7 @@ limitations under the License.
 #include "blueoil_opencv.hpp"
 
 namespace blueoil {
+
```
Comment (tkng) [inline runtime/src/blueoil_opencv.cpp]: it looks better to remove this blank line from the point of view of consistency.

**Comment by tkng** [comment]
Other than the small comments above, LGTM.

File: runtime/src/blueoil_opencv.cpp (hunk 1 above)
Comment (yoya) [reply runtime/src/blueoil_opencv.cpp]: Probably, We shold be revise runtime/README.md slightly to import the results of the dll build.

Comment (yoya) [reply runtime/src/blueoil_opencv.cpp]: I added the DLK library import chapter to runtime/README.md . I'm poor english writer. Perhaps, we need to re-write the runtime/README.md .

File: runtime/examples/CMakeLists.txt [hunk 2]
```diff
@@ -13,4 +13,26 @@ link_directories(${CMAKE_CURRENT_SOURCE_DIR}/lib)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 add_executable(a.out run.cpp)
-target_link_libraries(a.out blueoil pthread)
+
+if(USE_OPENCV)
+  find_package( OpenCV REQUIRED )
```
Comment (tkng) [inline runtime/examples/CMakeLists.txt]: Could you remove the spaces next to the parenthesis?

File: runtime/examples/CMakeLists.txt [hunk 3]
```diff
@@ -13,4 +13,26 @@ link_directories(${CMAKE_CURRENT_SOURCE_DIR}/lib)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 add_executable(a.out run.cpp)
```
Comment (tkng) [inline runtime/examples/CMakeLists.txt]: Is it possible to change the this binary name (`a.out`) to more self explainable name?

File: runtime/examples/run.cpp [hunk 4]
```diff
@@ -16,27 +16,34 @@ limitations under the License.
...
-    std::vector<float>& data = t.data();
-    unsigned seed = 0;
 
-    for (size_t i = 0; i < data.size(); ++i) {
-        const float f_rand = static_cast <float> (rand_r(&seed)) / static_cast <float> (RAND_MAX) * 255;
-        data[i] = f_rand;
+int main(int argc, char **argv) {
+  std::string imagefile, meta_yaml;
+
+  for (int i = 1 ; i < (argc-1); i++) {
```
Comment (tkng) [inline runtime/examples/run.cpp]: Could you delete the space to the left of the `;`?

Also, could you make the same modification for all for loops below?

Comment (yoya) [reply runtime/examples/run.cpp]: https://github.com/blue-oil/blueoil/pull/577/commits/ac1e423c1bc34800803e1a63ae81e47d1c851fd4

File: runtime/examples/CMakeLists.txt (hunk 3 above)
Comment (yoya) [reply runtime/examples/CMakeLists.txt]: https://github.com/blue-oil/blueoil/pull/577/commits/506e97a20f1be59e596f4855a61ab3e2c4cd2196

File: runtime/examples/CMakeLists.txt (hunk 2 above)
Comment (yoya) [reply runtime/examples/CMakeLists.txt]: https://github.com/blue-oil/blueoil/pull/577/commits/506e97a20f1be59e596f4855a61ab3e2c4cd2196

File: runtime/src/blueoil_opencv.cpp (hunk 1 above)
Comment (yoya) [reply runtime/src/blueoil_opencv.cpp]: https://github.com/blue-oil/blueoil/pull/577/commits/54f68cd5894fd12b80a5053405780e0c004c79c5
https://github.com/blue-oil/blueoil/pull/577/commits/15130c121ba74f8da168cd4c12df4a6851e0062b

**Pull request merged**